agent/submatview/store.go

```diff
@@ -0,0 +1,237 @@
+package submatview
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/consul/agent/cache"
+	"github.com/hashicorp/consul/lib/ttlcache"
+)
+
+// Store of Materializers. Store implements an interface similar to
+// agent/cache.Cache, and allows a single Materliazer to fulfill multiple requests
+// as long as the requests are identical.
+// Store is used in place of agent/cache.Cache because with the streaming
+// backend there is no longer any need to run a background goroutine to refresh
+// stored values.
+type Store struct {
+	logger     hclog.Logger
+	lock       sync.RWMutex
+	byKey      map[string]entry
+	expiryHeap *ttlcache.ExpiryHeap
```

Comment:
Perhaps this would be a good place to note that this expiryHeap unlike the one in agent/cache only contains _inactive_ entries? Might help orient folks who are expecting this to work the same way?